OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -133,7 +133,15 @@ public AssetBrowserLogic(Widget widget, Action onExit, ModData modData, WorldRen
 
 			filenameInput = panel.Get<TextFieldWidget>("FILENAME_INPUT");
 			filenameInput.OnTextEdited = () => ApplyFilter();
-			filenameInput.OnEscKey = _ => filenameInput.YieldKeyboardFocus();
+			filenameInput.OnEscKey = _ =>
+			{
+				if (string.IsNullOrEmpty(filenameInput.Text))
+					filenameInput.YieldKeyboardFocus();
+				else
+					filenameInput.Text = "";
```

Comment:
Also need to call `ApplyFilter` to actually reset this.

Reply:
How about `filenameInput.OnTextEdited()` (it calls `ApplyFilter`)?

Reply:
Sure